### PR TITLE
load model-configuration from file before creating the test mode

### PR DESCRIPTION
**Pull request by addyess**

### CODE REVIEW

File: pytest_operator/plugin.py [hunk 1]
```diff
@@ -77,6 +77,13 @@ def pytest_addoption(parser):
         help="Store the completed crash dump in this dir. "
         "The default is current folder.",
     )
+    parser.addoption(
+        "--model-config",
+        action="store",
+        default="model_config.yaml",
```
Comment (Cynerva) [inline pytest_operator/plugin.py:83]: I would default this to `None` or `""`. Better not to accidentally load some file lying around; the user should have to pass it in explicitly.

File: pytest_operator/plugin.py [hunk 2]
```diff
@@ -406,8 +414,17 @@ async def _setup_model(self):
             await self._controller.connect(self.controller_name)
             on_cloud = f" on cloud {self.cloud_name}" if self.cloud_name else ""
             log.info(f"Adding model {self.model_full_name}{on_cloud}")
+
+            model_config_file = Path(self.model_config)
+            model_config = None
+            if not model_config_file.exists():
+                log.warning("Can't apply model-config from %s", model_config_file)
```
Comment (Cynerva) [inline pytest_operator/plugin.py:421]: If the user has specified a `--model-config` file, but the file couldn't be found, they will probably want to know about it. I would go further and log this message at the error level, or maybe even halt execution by raising an Exception.

If the user has *not* specified a `--model-config` file, it seems better to just not try to load it. No need for a warning in that case.

**Review by Cynerva — APPROVED** [review]
I think the default behavior when `--model-config` is unspecified could be improved. See my other comments. It's not really a show-stopper, so I'm approving this. I'll leave it up to you if you want to merge as-is.

File: pytest_operator/plugin.py (hunk 2 above)
Comment (addyess) [reply pytest_operator/plugin.py]: agree, things always make more sense after a day.  great feedback.  I'll update to this

**Pull request merged**